File: src/main/java/com/wxl/dyttcrawler/core/ThreadPool.java
package com.wxl.dyttcrawler.core;

import java.util.concurrent.Future;

/**
 * Create by wuxingle on 2020/6/26
 * 爬虫线程池
 */
public interface ThreadPool {

    /**
     * 任务提交
     * 阻塞到提交成功
     */
    Future<?> execute(Runnable runnable) throws InterruptedException;

    /**
     * 关闭
     * 阻塞到关闭结束
     */
    void shutdown() throws InterruptedException;

    /**
     * 是否关闭
     */
    boolean isShutDown();


    /**
     * 运行中线程
     */
    int getActiveThreadNum();

    /**
     * 总线程
     */
    int getMaxThreadNum();
}

File: src/main/kotlin/com/wxl/dyttcrawler/core/ThreadPool.kt
package com.wxl.dyttcrawler.core

import java.util.concurrent.Future

/**
 * Create by wuxingle on 2021/10/08
 * 爬虫线程池
 */
interface ThreadPool {

    /**
     * 是否关闭
     */
    val isShutDown: Boolean

    /**
     * 运行中线程数
     */
    val activeThreadNum: Int

    /**
     * 总线程数
     */
    val maxThreadNum: Int

    /**
     * 任务提交
     * 阻塞到提交成功
     */
    @Throws(InterruptedException::class)
    fun execute(runnable: Runnable): Future<*>

    /**
     * 关闭
     * 阻塞到关闭结束
     */
    @Throws(InterruptedException::class)
    fun shutdown()
}